crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -653,8 +653,26 @@ where
 }
 
 fn expression() -> impl ExprParser {
-    recursive(|expr| expression_with_precedence(Precedence::Lowest, expr, false))
-        .labelled("expression")
+    recursive(|expr| {
+        choice((
+            return_expression(expr.clone()),
```

Comment:
I think we should parse returns in the `statement` parser rather than here

Reply:
@jfecher The Rust language permits returns in expression atom position. Should I make it a statement anyway?